src/types/CompleteEventsHandler.ts

```diff
@@ -0,0 +1,33 @@
+import { EventInterface, EventHandler } from './Event';
+
+export enum CompleteEventsPhase {
```

Comment:
I'm not sure I understand the purpose of this enum? 

As far as enumerations go, in my mind it would be more intuitive to have an enum with different event types as variants, so it can be used to branch code (if, switch etc) on the enum's values.
```ts
enum Phases {
  CompleteEvent = 'X'
  InstantEvent = 'I'
}
```

Reply:
So, each event has different phases, and I guess it does not make sense to have that enum for events with a singular phase, I implemented that just to stay consistent throughout all the events, but you are right, in the case of certain events like Complete Event, Instant Event etc (events with only one value of phase), it doesn't make sense to have enums